pallets/multisig/src/lib.rs

```diff
@@ -858,68 +920,74 @@ impl<T: Trait> Module<T> {
             !Self::votes(&multisig_signer_proposal),
             Error::<T>::AlreadyVoted
         );
-        if let Some(proposal) = Self::proposals(&multisig_proposal) {
-            let mut proposal_details = Self::proposal_detail(&multisig_proposal);
-            proposal_details.approvals += 1u64;
-            let current_did = Context::current_identity::<Identity<T>>().unwrap_or_default();
-            match proposal_details.status {
-                ProposalStatus::Invalid => return Err(Error::<T>::ProposalMissing.into()),
-                ProposalStatus::Rejected => return Err(Error::<T>::ProposalAlreadyRejected.into()),
-                ProposalStatus::ExecutionSuccessful | ProposalStatus::ExecutionFailed => {}
-                ProposalStatus::ActiveOrExpired => {
-                    // Ensure proposal is not expired
-                    if let Some(expiry) = proposal_details.expiry {
-                        ensure!(
-                            expiry > <pallet_timestamp::Module<T>>::get(),
-                            Error::<T>::ProposalExpired
-                        );
-                    }
-                    Self::execute_proposal(
+        ensure!(
+            <Proposals<T>>::contains_key(&multisig_proposal),
+            Error::<T>::ProposalMissing
+        );
+
+        let mut proposal_details = Self::proposal_detail(&multisig_proposal);
+        proposal_details.approvals += 1u64;
+        let multisig_did = <MultiSigToIdentity<T>>::get(&multisig);
+        match proposal_details.status {
+            ProposalStatus::Invalid => return Err(Error::<T>::ProposalMissing.into()),
+            ProposalStatus::Rejected => return Err(Error::<T>::ProposalAlreadyRejected.into()),
+            ProposalStatus::ExecutionSuccessful | ProposalStatus::ExecutionFailed => {}
```

Comment:
Should we return an error (`AlreadyExecuted`) in these cases to avoid updating the state with irrelevant data (e.g. incrementing approvals / votes)?